_nix

```diff
@@ -978,8 +977,39 @@ function _nix_completion () {
                   ${nixops_common_arguments[*]} \
                   ${main_options[*]} \
                   ${nix_boilerplate_opts[*]} && return 0
+
+            local statecmd
+            if [[ "${opts[--state]}" || "${opts[-s]}" ]]; then
+                local statefile="${opt_args[--state]:-${opt_args[-s]}}"
+                if [[ "$statefile" == *.nixops && -f "$statefile" ]]; then
+                    statecmd="--state $statefile"
+                fi
+            fi
+
+            local deploymentcmd
+            if [[ "${opts[--deployment]}" || "${opts[-d]}" ]]; then
+                local deployment="${opt_args[--deployment]:-${opt_args[-d]}}"
+                deploymentcmd="--deployment $deployment"
+            fi
+
+            case "$state" in
+                nixops_deployments)
+                    local -a deployments=()
+                    while read -r _ id _ name _; do deployments+=("$id" "$name")
+                    done < <(nixops list $statecmd | tail -n +4 | head -n -1)
```

Comment:
I don't like having to parse command output like this. It's needed because nixops returns things in pretty tables. As far as I could tell, there's no "plumbing" versions of their commands.